lua/mini/extra.lua

```diff
@@ -1725,6 +1786,24 @@ H.choose_with_buflisted = function(item)
   vim.bo[buf_id].buflisted = true
 end
 
+-- Colorscheme picker ----------------------------------------------------------
+H.get_colorscheme = function()
+  local has_colors, colors = pcall(require, 'mini.colors')
+  return has_colors and colors.get_colorscheme() or vim.g.colors_name
+end
+
+H.set_colorscheme = function(colorscheme, background)
+  vim.o.background = background
+  vim.schedule(function()
+    if type(colorscheme) == 'string' then
+      vim.cmd('colorscheme ' .. colorscheme)
+    else
+      colorscheme:apply()
+      vim.cmd('doautocmd ColorScheme') -- trigger for things like mini.misc termbg_sync
+    end
+  end)
+end
```

Comment:
```suggestion
H.set_colorscheme = vim.schedule_wrap(function(colorscheme, background)
  vim.o.background = background
  if colorscheme == vim.g.colors_name then return end
  if type(colorscheme) == 'string' then return vim.cmd('colorscheme ' .. colorscheme) end
  colorscheme:apply()
  -- Trigger to indicate actual color scheme application
  vim.cmd('doautocmd ColorScheme')
end)
```

This can be simplified. There shouldn't be any harm in doing this all inside `vim.schedule()`.

Reply:
Thanks. I did not know about `schedule_wrap`.

To be honest, I don't really have a good intuition about when one needs to schedule something or not. For me, it's one of those things where if weird stuff happens, try schedule.

Reply:
> For me, it's one of those things where if weird stuff happens, try schedule.

Seems pretty accurate to me :)